### PR TITLE
Adds test data base path to classic test configs for more easily switching between test data location

### PR DESCRIPTION
**Pull request by jfy133**
https://nfcore.slack.com/archives/C043UU89KKQ/p1713940102697309

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Documentation in `docs` is updated

### CODE REVIEW

**Comment by mashehu** [comment]
Is there documentation about this somewhere?

**Comment by maxulysse** [comment]
> Is there documentation about this somewhere?

I'm on it

**Comment by jfy133** [comment]
> Is there documentation about this somewhere?

What do you mean? It's only really aimed at developers I guess? And I updated the schema...

File: nf_core/pipeline-template/nextflow_schema.json [hunk 1]
```diff
@@ -267,6 +267,12 @@
                     "description": "Validation of parameters in lenient more.",
                     "hidden": true,
                     "help_text": "Allows string values that are parseable as numbers or booleans. For further information see [JSONSchema docs](https://github.com/everit-org/json-schema#lenient-mode)."
+                },
+                "test_data_base_path":{
+                    "type": "string",
+                    "fa_icon": "far fa-check-circle",
+                    "description": "Base URL to location of pipeline test dataset files",
```
Comment (jfy133) [inline nf_core/pipeline-template/nextflow_schema.json:274]:
```suggestion
                    "description": "Base URL or local path to location of pipeline test dataset files",
```

File: nf_core/pipeline-template/nextflow_schema.json [hunk 2]
```diff
@@ -267,6 +267,12 @@
                     "description": "Validation of parameters in lenient more.",
                     "hidden": true,
                     "help_text": "Allows string values that are parseable as numbers or booleans. For further information see [JSONSchema docs](https://github.com/everit-org/json-schema#lenient-mode)."
+                },
+                "test_data_base_path":{
```
Comment (jfy133) [inline nf_core/pipeline-template/nextflow_schema.json:271]:
```suggestion
                "test_data_base_path": {
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@
 - Update python to 3.12 ([#2805](https://github.com/nf-core/tools/pull/2805))
 - Remove `pyproject.toml` from template root
 - Shorten lines in pipeline template ([#2908](https://github.com/nf-core/tools/pull/2908))
+- Add a new hidden `--test_data_base_path` parameter to more easily switch locations of test data in test configs (#2900)[]
```
Comment (maxulysse) [inline CHANGELOG.md:16]:
```suggestion
- Add a new hidden `--pipelines_testdata_base_path` parameter to more easily switch locations of test data in test configs (#2900)[]
```

File: nf_core/pipeline-template/conf/test.config [hunk 4]
```diff
@@ -22,7 +22,7 @@ params {
     // Input data
     // TODO nf-core: Specify the paths to your test data on nf-core/test-datasets
     // TODO nf-core: Give any required params for the test so that command line flags are not needed
-    input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
+    input  = test_data_base_path + 'viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
```
Comment (maxulysse) [inline nf_core/pipeline-template/conf/test.config:25]:
```suggestion
    input  = pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
```

File: nf_core/pipeline-template/conf/test_full.config [hunk 5]
```diff
@@ -17,13 +17,13 @@ params {
     // Input data for full size test
     // TODO nf-core: Specify the paths to your full test data ( on nf-core/test-datasets or directly in repositories, e.g. SRA)
     // TODO nf-core: Give any required params for the test so that command line flags are not needed
-    input = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
+    input = test_data_base_path + 'viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
```
Comment (maxulysse) [inline nf_core/pipeline-template/conf/test_full.config:20]:
```suggestion
    input = pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
```

File: nf_core/pipeline-template/conf/test_full.config [hunk 6]
```diff
@@ -17,13 +17,13 @@ params {
...
 
     {% if igenomes -%}
     // Genome references
     genome = 'R64-1-1'
     {%- else -%}
     // Fasta references
-    fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/genome/NC_045512.2/GCF_009858895.2_ASM985889v3_genomic.200409.fna.gz'
+    fasta = test_data_base_path + 'viralrecon/genome/NC_045512.2/GCF_009858895.2_ASM985889v3_genomic.200409.fna.gz'
```
Comment (maxulysse) [inline nf_core/pipeline-template/conf/test_full.config:27]:
```suggestion
    fasta = pipelines_testdata_base_path + 'viralrecon/genome/NC_045512.2/GCF_009858895.2_ASM985889v3_genomic.200409.fna.gz'
```

File: nf_core/pipeline-template/nextflow_schema.json [hunk 7]
```diff
@@ -267,6 +267,12 @@
                     "description": "Validation of parameters in lenient more.",
                     "hidden": true,
                     "help_text": "Allows string values that are parseable as numbers or booleans. For further information see [JSONSchema docs](https://github.com/everit-org/json-schema#lenient-mode)."
+                },
+                "test_data_base_path": {
```
Comment (maxulysse) [inline nf_core/pipeline-template/nextflow_schema.json:271]:
```suggestion
                "pipelines_testdata_base_path": {
```

**Review by maxulysse — COMMENTED** [review]
need to tinker the path a bit more

File: nf_core/pipeline-template/nextflow.config [hunk 8]
```diff
@@ -37,6 +37,7 @@ params {
     hook_url                   = null
     help                       = false
     version                    = false
+    test_data_base_path        = 'https://raw.githubusercontent.com/nf-core/test-datasets/'
```
Comment (jfy133) [inline nf_core/pipeline-template/nextflow.config:40]:
```suggestion
    pipelines_testdata_base_path        = 'https://raw.githubusercontent.com/nf-core/test-datasets/'
```

**Review by mirpedrol — APPROVED** [review]
Is this ready to be merged?

File: nf_core/pipeline-template/conf/test.config [hunk 9]
```diff
@@ -22,7 +22,7 @@ params {
     // Input data
     // TODO nf-core: Specify the paths to your test data on nf-core/test-datasets
     // TODO nf-core: Give any required params for the test so that command line flags are not needed
-    input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
+    input  = pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
```
Comment (maxulysse) [inline nf_core/pipeline-template/conf/test.config:25]:
```suggestion
    input  = params.pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv'
```

File: nf_core/pipeline-template/conf/test_full.config [hunk 10]
```diff
@@ -17,13 +17,13 @@ params {
     // Input data for full size test
     // TODO nf-core: Specify the paths to your full test data ( on nf-core/test-datasets or directly in repositories, e.g. SRA)
     // TODO nf-core: Give any required params for the test so that command line flags are not needed
-    input = 'https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
+    input = pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
```
Comment (maxulysse) [inline nf_core/pipeline-template/conf/test_full.config:20]:
```suggestion
    input = params.pipelines_testdata_base_path + 'viralrecon/samplesheet/samplesheet_full_illumina_amplicon.csv'
```

**Pull request merged**